airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -258,15 +262,32 @@ def error_message(self, response: requests.Response) -> str:
         """
         return ""
 
+    def must_deduplicate_query_params(self) -> bool:
+        return False
+
+    def deduplicate_query_params(self, url: str, params: Optional[Mapping[str, Any]]) -> Mapping[str, Any]:
+        if params is None:
+            params = {}
+        query_string = urllib.parse.urlparse(url).query
+        query_dict = {k: v[0] for k, v in urllib.parse.parse_qs(query_string).items()}
+
+        duplicate_keys_with_same_value = {k for k in query_dict.keys() if params.get(k) == query_dict[k]}
+        return {k: v for k, v in params.items() if k not in duplicate_keys_with_same_value}
+
     def _create_prepared_request(
```

Comment:
This method seems to be [a duplicate of the newly created class](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py#L353-L373). Should we update this as well?

Reply:
yes, good idea